### ISSUE TITLE
ENCRYPTABLE SHARING: Credential sharing

### ISSUE THREAD

**Issue by lkleisa** [puzzle/cryptopus#688]
In [Ticket number 657](https://github.com/puzzle/cryptopus/issues/657) and [in this PR](https://github.com/puzzle/cryptopus/pull/658/), we implemented the feature for secure sharing files in cryptopus. Now we want to add the possibility for sharing credentials with a user. 

The frontend should now have a new button in the `encryptable show` component. 

![Image](https://user-images.githubusercontent.com/68543649/235473696-4da15832-2068-495a-867e-d833b5df473b.png)

This share button opens a new dialog like the transfer file dialog. It has one input field with all possible candidates (all users except self). This already implemented in `transfer file` component

![Image](https://user-images.githubusercontent.com/68543649/235473857-ce156267-0efe-4961-99e5-e5aae4d7ea29.png)

Similar to the file sharing feature, the shared credentials should be transferred to the inbox folder in the personal team of the receiver.

![Image](https://user-images.githubusercontent.com/68543649/235474081-1b5b8518-3a24-4c8f-85db-11be4f5be76b.png)

All the en-, de- and recryption can be copied from the file sharing feature (EncryptableTransferController). 

Clarify the following:
- Should there be another view for the inbox folder, so that files and credentials can be distinguished?
- Should it be possible to share a transferred encryptable?
- Should it be possible to share an encryptable from a private team?

TODO:
- [x] Clarify the above points
- [x] Implement the sharing dialog
- [x] Implement recryption and saving into the receiver inbox folder, same as file sharing
- [x] Adjust the `encryptable row` component with the new type of encryptable
- [x] Frontend Testing
- [x] Backend Testing
- [x] Follow-up ticket for moving transferred credentials in another folder
- [x] Implement PR Feedback backend
- [x] Implement PR Feedback frontend
- [x] Implement possibility to edit transferred credentials
- [x] Implement possibility to move transferred credentials to personal folder
- [x] Smoke testing
- [ ] Implement feedback

**Comment by lkleisa** [comment]
Way to go: 
- [ ] Add new share button to encryptable show
- [ ] Add new pop up for selecting the receiver
- [ ] Adjust the backend -> copy possible things from file sharing
- [ ] Look, that for file and credential the same way is used as often as possible

**Comment by lkleisa** [comment]
Is implemented in PR #695 and we need to wait on merging of file sharing, because this feature is based on file sharing

**Comment by lkleisa** [comment]
# To do
- [ ] Implement PR Feedback (only 3 - 4 open)
- [ ] Implement possibility to edit transferred credentials
- [ ] Implement possibility to move transferred credentials to personal folder
- [ ] Smoke testing

**Comment by lkleisa** [comment]
- [x] Fix rubocop
- [x] Make the increasement more readable 
- [x] Make the increasement working
- [ ] Smoke testing
